src/logging.cpp

```diff
@@ -12,6 +12,35 @@ namespace Generators {
 LogItems g_log;
 static std::ostream* gp_stream{&std::cerr};
 static std::unique_ptr<std::ofstream> gp_logfile;
+static CallbackFn gp_callback{};
+
+// Custom stream that calls gp_callback on every line of output
+struct CallbackStream : std::ostream {
+  CallbackStream() : std::ostream{&m_buffer} {
+    gp_stream = this;
+  }
+
+  struct CustomBuffer : std::stringbuf {
+    int sync() override {
+      auto string = str();
+      if (gp_callback)
+        gp_callback(string.c_str(), string.size());
+      str("");
+      return 0;
+    }
+  };
+
+  CustomBuffer m_buffer;
+} gp_callback_stream;
+
+void SetLogStream() {
+  if (gp_callback)
+    gp_stream = &gp_callback_stream;
+  else if (gp_logfile)
+    gp_stream = gp_logfile.get();
```

Comment:
what if someone sets both a log file and a callback? I think it would make sense to have the last one set win, log to both, or be an error. I guess this precedence is fine, but it should be documented.

Reply:
Good point. There's no way to set a non filename log, it's only reset by setting an empty filename. So it makes sense to work through priorities. Where a filename takes precedence over cerr and a callback takes priority over a filename. I'll update the documentation to make it clear.

Reply:
Updated the logic so now the most recent logging destination will be selected, and other destinations will be removed/reset.